docs/source/complex_numbers.rst

```diff
@@ -0,0 +1,174 @@
+.. _complex_numbers-doc:
+
+Complex Numbers
+===============
+
+Complex numbers are numbers that can be expressed in the form :math:`a + bj`, where a and b are real numbers,
+and *j* is a solution of the equation :math:`x^2 = −1`. Complex numbers frequently occur in mathematics and
+engineering, especially in signal processing. Tensors of complex dtypes provide a more natural user experience
+for users and libraries (eg. TorchAudio) that previously worked around the lack of complex tensors by using
+float tensors with shape :math:`(..., 2)` where the last dimension contained the real and imaginary values.
+
+Operations on complex tensors (eg :func:`torch.mv`, :func:`torch.matmul`) are likely to be faster and more
+memory efficient than operations on float tensors mimicking them. Operations involving complex numbers in
+PyTorch are optimized to use vectorized assembly instructions and specialized kernels (e.g. LAPACK, CuBlas).
+Thus using functions for complex tensors will provide performance benefits as opposed to users defining
+their own functions.
+
+.. note::
+     Spectral Ops currently don't use complex tensors but the API would be soon updated to use complex tensors.
+
+.. warning ::
+     Complex Tensors is a beta feature and subject to change.
+
+Creating Complex Tensors
+------------------------
+
+We support two complex dtypes: `torch.cfloat` and `torch.cdouble`
+
+::
+
+     >>> x = torch.randn(2,2, dtype=torch.cfloat)
+     >>> x
+     tensor([[-0.4621-0.0303j, -0.2438-0.5874j],
+          [ 0.7706+0.1421j,  1.2110+0.1918j]])
+
+.. note::
+
+     The default dtype for complex tensors is determined by the default floating point dtype.
+     If the default floating point dtype is torch.float64 then complex numbers are inferred to
+     have a dtype of torch.complex128, otherwise they are assumed to have a dtype of torch.complex64.
+
+All factory functions apart from :func:`torch.linspace`, :func:`torch.logspace`, and :func:`torch.arange` are
+supported for complex tensors.
+
+Transition from the old representation
+--------------------------------------
+
+Users who currently worked around the lack of complex tensors with real tensors of shape `(..., 2)`
+can easily to switch using the complex tensors in their code using :func:`torch.view_as_complex` and
+- :func:`torch.view_as_real`. Note that these functions don't perform any copy and
+return a view of the input Tensor.
+
+::
+
+     >>> x = torch.randn(3, 2)
+     >>> x
+     tensor([[ 0.6125, -0.1681],
+          [-0.3773,  1.3487],
+          [-0.0861, -0.7981]])
+     >>> y = torch.view_as_complex(x)
+     >>> y
+     tensor([ 0.6125-0.1681j, -0.3773+1.3487j, -0.0861-0.7981j])
+     >>> torch.view_as_real(y)
+     tensor([[ 0.6125, -0.1681],
+          [-0.3773,  1.3487],
+          [-0.0861, -0.7981]])
+
+Accessing real and imag
+-----------------------
+
+The real and imaginary values of a complex tensor can be accessed using the :attr:`real` and
+:attr:`imag` views.
```

Comment:
nit: I think we talked about this in other parts of the documentation, but it is relying very hard on the reader's comprehension skills to only say "view" here, and expect them to unfold this into "accessing these attributes doesn't allocate any memory, and you can do in-place updates on them which will update the original complex tensor, and by the way, these tensors are not contiguous"